plugins/bitrot/rest.go

```diff
@@ -177,5 +177,67 @@ func bitrotDisableHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "bitrot Disable")
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "Bitrot disabled successfully")
+}
+
+func bitrotScrubOndemandHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volName := p["volname"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if bitrot is disabled
+	val, exists := volinfo.Options[volume.VkeyFeaturesBitrot]
+	if exists && val == "off" {
```

Comment:
This condition is wrong when bitrot is not enabled. (exists will be false). Better make this as function in volume as `volume.IsBitrotEnabled`. since it is used in multiple places now.

Reply:
Done